app/src/main/java/org/mozilla/focus/broadcastreceiver/DownloadBroadcastReceiver.java

```diff
@@ -63,13 +63,14 @@ private void displaySnackbar(final Context context, long completedDownloadRefere
                     final String localUri = uriString.startsWith(FILE_SCHEME) ? uriString.substring(FILE_SCHEME.length()) : uriString;
                     final String fileExtension = MimeTypeMap.getFileExtensionFromUrl(localUri);
                     final String mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(fileExtension);
-                    final String fileName = URLUtil.guessFileName(localUri, null, mimeType);
-
-                    final File file = new File(localUri);
-                    final Uri uriForFile = FileProvider.getUriForFile(context, BuildConfig.APPLICATION_ID + FILE_PROVIDER_EXTENSION, file);
-
-                    final Intent openFileIntent = IntentUtils.createOpenFileIntent(uriForFile, mimeType);
-                    showSnackbarForFilename(openFileIntent, context, fileName);
+                    final String fileName = URLUtil.guessFileName(Uri.decode(localUri), null, mimeType);
+
+                    final File file = new File(Uri.decode(localUri));
+                    if (file.exists()) {
```

Comment:
I'd either remove this check - now that we expect this file to always exist.

Or: Throw an exception (like IllegalStateException) so that we see a crash and know that there are still issues.